WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationMLSMessageAddEventProcessor.swift

```diff
@@ -31,10 +33,138 @@ protocol ConversationMLSMessageAddEventProcessorProtocol {
 }
 
 struct ConversationMLSMessageAddEventProcessor: ConversationMLSMessageAddEventProcessorProtocol {
+    
+    enum Failure: Error {
+        case mlsConversationNotFound
+    }
+
+    let conversationLocalStore: any ConversationLocalStoreProtocol
+    let messageLocalStore: any MessageLocalStoreProtocol
+    let userLocalStore: any UserLocalStoreProtocol
+    let protobufMessageProcessor: any ConversationProtobufMessageProcessorProtocol
+
+    func processEvent(_ event: ConversationMLSMessageAddEvent) async throws {
+        let conversationID = event.conversationID
+        let senderID = event.senderID
+        let date = event.timestamp
+        let decryptedMessages = event.decryptedMessages
+        
+        guard !decryptedMessages.isEmpty else {
+            return WireLogger.proteus.warn(
+                "failed to add MLS message: there are no decrypted messages to process"
+            )
+        }
+
+        for decryptedMessage in decryptedMessages {
+            try await processDecryptedMessage(
+                decryptedMessage,
+                conversationID: conversationID,
+                senderID: senderID,
+                date: date
+            )
+        }
+    }
+
+    private func processDecryptedMessage(
+        _ decryptedMessage: ConversationMLSMessageAddEvent.DecryptedMessage,
+        conversationID: ConversationID,
+        senderID: UserID,
+        date: Date?
+    ) async throws {
+        guard let conversation = await conversationLocalStore.fetchConversation(
+            id: conversationID.uuid,
+            domain: conversationID.domain
+        ) else {
+            throw Failure.mlsConversationNotFound
+        }
+
+        let logAttributes: LogAttributes = [
+            .messageType: "conversation.mls-message-add",
```

Comment:
not a blocker for now, but I would expect to rely on a exhaustive list of values for messageType instead of any String. So porting an enum to either WireAPI or WireDomain